sql/persisted_variable.cc

```diff
@@ -2094,6 +2115,27 @@ bool Persisted_variables_cache::reset_persisted_variables(THD *thd,
   } else {
     auto erase_variable = [&](const char *name_cptr) -> bool {
       string name_str{tolower_varname(name_cptr)};
+#ifdef WITH_WSREP
+      /* Some of the wsrep_variables both have
+       * 1. uppercase characters in them
+       * 2. aliases
+       *
+       * e.g:
+       * wsrep_slave_FK_checks, wsrep_applier_FK_checks
+       * wsrep_slave_UK_checks, wsrep_applier_UK_checks
+       *
+       * SET PERSIST of these variables can result in multiple entries in the
+       * performance_schema.persisted_variables. After server restart, any
+       * attempt to RESET PERSIST of these variables may fail the operation and
+       * result in an inconsistent state.
+       *
+       * So, temporarily we skip converting "wsrep_" variables to lowercase
+       * before persisting them. */
+      const char wsrep_prefix[]{"wsrep_"};
```

Comment:
static constexpr?

Or even better (name_str is std::string) something like:
```
static const std::string wsrep_prefix("wsrep_");
if (!name_str.rfind(wsrep_prefix, 0)) {
  // name starts with wsrep_
}
```
or
```
static const std::string wsrep_prefix("wsrep_");
if (!name_str.compare(o, wsrep_prefix.size(), wsrep_prefix)) {
  // name starts with wsrep_
}

```

Reply:
Changed to 
```
+      static const std::string wsrep_prefix("wsrep_");
+      if (!name_str.rfind(wsrep_prefix, 0)) {
+        // name starts with wsrep_
+        name_str.assign(name_cptr);
+      }
```